### PR TITLE
feat: add assign to project with fields

### PR DESCRIPTION
**Pull request by luizstacio**
- Add action to assign new issues and pull requests to a project
- Add docs and licenses
- Add CI to run lint and test on assign-project
- Add conditional CI to just run if files under assign-project was changed

### CODE REVIEW

**Review by ellioty — DISMISSED** [review]

**Comment by Voxelot** [comment]
@luizstacio Would it make sense to mock the GH api and have some basic unit tests to ensure the values are set correctly from the action inputs?

**Comment by luizstacio** [comment]
> @luizstacio Would it make sense to mock the GH api and have some basic unit tests to ensure the values are set correctly from the action inputs?

I think it makes sense, I was looking to have tests in the CI level also, can we implement it on a next issue?

**Comment by luizstacio** [comment]
> @luizstacio Would it make sense to mock the GH api and have some basic unit tests to ensure the values are set correctly from the action inputs?

I'm going to add some tests in this way

File: .github/workflows/ci-assign-project.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+on:
+  pull_request:
+    types: [opened, synchronize, edited, closed]
+
+jobs:
+  cancel-previous-run:
+    name: Cancel previous actions
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (Voxelot) [inline .github/workflows/ci-assign-project.yml:12]: you shouldn't need a git checkout for this job

File: .github/workflows/ci-assign-project.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      - uses: n1hility/cancel-previous-runs@v2
```
Comment (Voxelot) [inline .github/workflows/ci-assign-project.yml:13]: The recommended way to accomplish this now is via the `concurrency` setting:

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-concurrency-to-cancel-any-in-progress-job-or-run

It seems like the best way to configure concurrency is described here: https://github.com/celestiaorg/celestia-node/pull/894#discussion_r952863065

File: gh-projects/assign-to-project/src/actions/assignProject.test.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
+import * as octokitGraphql from '@octokit/graphql';
```
Comment (Voxelot) [inline gh-projects/assign-to-project/src/actions/assignProject.test.ts:1]: Is this test run in CI? I only see `lint` and `build` steps there.

Comment (luizstacio) [reply gh-projects/assign-to-project/src/actions/assignProject.test.ts:1]: The test is inside the build step I will rename it to test

**Review by Voxelot — APPROVED** [review]

**Pull request merged**